File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque

def solution(cacheSize, cities):
    answer = 0
    cache = deque(maxlen=cacheSize)
    # 캐시 크기가 0일 때
    if cacheSize == 0:  
        return len(cities) * 5

    for c in cities:
        # 대소문자 구분하지 않기 위해 모두 소문자로 변경
        c = c.lower()  
        # 캐시에 있는 데이터라면
        if c in cache:  
            answer += 1
             # 데이터 삭제
            cache.remove(c) 
        # 캐시에 없는 데이터라면
        else:  
            answer += 5
        cache.append(c)
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    const cache = []
    var answer = 0;
    
    for(let city of cities){
        // 대소문자 구분 없음
        city = city.toLowerCase() 
        
        const idx = cache.indexOf(city)
        
        // cache hit
        if(idx!==-1){
            cache.splice(idx,1) // 해당 도시 제거
            cache.push(city) // 뒤에 추가
            answer+=1            
        }else{ 
        // cache miss
            // cache에 빈 공간이 없는 경우,
            if(cacheSize && cache.length >= cacheSize){
                cache.shift() // 가장 오래된 도시 제거
            }
            
            // cahce에 도시 넣기
            if(cacheSize) cache.push(city)
            answer += 5            
        }
    }
    
    return answer;
}